### DISCUSSION TITLE
Refactoring structures

### DISCUSSION THREAD

**Discussion by Seggan** [Vyxal/Vyxal#744] [Version 3]
We had a [discussion](https://chat.stackexchange.com/transcript/message/60483383#60483383) in chat about the possibility of changing the characters used for logical structures and literals. The fact that lists use `⟨⟩` for their syntax makes typing hard. I find myself constantly copy pasting single characters when writing Vyxal code or developing Jyxal. Using backticks for strings also likes to mess with Markdown. After talking about it for a while, me, @SE-Nitro, and @cgccuser (hopefully that's the correct usernames) decided to propose using more "common" characters for these structures. Taking some insight from the [corpus](https://gist.github.com/chunkybanana/ab27c90a938dfcd215081a6e1b3cff5c), we decided to do away with pipes and replace them with commas. Strings would also be delimited by `"` instead of argh-I-can't-write-a-single-backtick-in-markdown. So the structures themselves would become: `[a, normal, perfectly, sensible, list]`, `{if, else}`, `(variable, for body)`, `⟨cond, while body⟩`, and `"a normal string"`. What da ya think?

**Comment by mathcat4** [comment]
TBH, I fully approve of using `[]` for lists and the other stuff, but I would prefer pipes because it's taller than a comma, and sort of divides the parts.
Example: `{thisisareallylongifpart,andthisistheelsepart}` vs. `{thisisareallylongifpart|andthisistheelsepart}`
<sup>Okay, you might not see the difference</sup>
Although, it might not be useful when printing the python/traditional variant or working with it.

**Reply by Seggan** [reply]
I don't mind keeping the pipes, we'd need something to replace the comma element if we did use commas.

**Reply by Seggan** [reply]
Also, functions and lambdas use pipes as well, so if we moved to commas we'd have to replace those as well

**Reply by cgccuser** [reply]
Yeah, having to use commas instead of pipes for every structure is probably the main concern about this. IMO it's fine but let's see what others think.

**Reply by a-stone-arachnid** [reply]
My opinion: pipes for separators, `'single quotes'` for strings, the rest looks good to me.

**Reply by lyxal** [reply]
On the topic of lists, my idea is to have `#[`  and `#]` be digraphs that are aliases for `⟨` and `⟩`.

On the topic of strings, the whole point of having backticks is to be different from 99% of golfing languages. Also, `'` isn't a suitable character because that'll be for one-character strings (backslash will be for reduction instead).

On the topic of pipes and commas, pipes, IMO, are clearer and more visually distinct than commas.

**Reply by Seggan** [reply]
> On the topic of strings, the whole point of having backticks is to be different from 99% of golfing languages. Also, ' isn't a suitable character because that'll be for one-character strings (backslash will be for reduction instead).

But is not the whole point of Vyxal to be more like conventional languages? Backticks are terribly inconvenient if posting code examples. Why *should* we be different than Husk and 05AB1E in this aspect? Strings are an integral part of any programming language (aside from Jelly).

**Reply by lyxal** [reply]
Here's something backtick strings allow: easy inclusion of `"` without having to escape `"`.  And I'm pretty sure I've used that fact somewhere before.

**Reply by lyxal** [reply]
However, forgive me for being what may seem stubborn - major changes like these take me a while to adjust to, and language changes aren't something I'm too used to.

**Reply by pxeger** [reply]
> Here's something backtick strings allow: easy inclusion of `"` without having to escape `"`

Here's something double-quoted strings allow: easy inclusion of <code>&#96;</code> without having to escape <code>&#96;</code>

The best delimiters for strings are `«` `»` IMO, because they're easy to type on many international keyboard layouts, but don't require escaping of *any* ASCII characters.

**Reply by lyxal** [reply]
The only problem with `«` and `»` for strings is that they're non-ascii, which might not be the most useful for things like restricted source.

**Reply by cgccuser** [reply]
@Lyxal I think that's fine, we could add digraphs for them.

**Comment by chunkybanana** [comment]
<code>&lt;code>`&lt;/code></code>

**Reply by pxeger** [reply]
?
